src/Controls/src/Core/MenuBar.cs

```diff
@@ -41,19 +37,24 @@ public void Add(IMenuBarItem item)
 			var index = _menus.Count;
 			_menus.Add(item);
 			NotifyHandler(nameof(IMenuBarHandler.Add), index, item);
+		}
 
-			// Take care of the Element internal bookkeeping
-			if (item is Element element)
+		internal void SyncMenuBarItemsFromPages(IList<MenuBarItem> menuBarItems)
+		{
+			for (int i = 0; i < menuBarItems.Count; i++)
 			{
-				OnChildAdded(element);
+				var menuBarItem = menuBarItems[i];
+				if (this.Count > i && this[i] == menuBarItem)
+					continue;
+
+				if (this.Contains(menuBarItem))
+					Remove(menuBarItem);
+
+				Insert(i, menuBarItem);
 			}
-		}
 
-		internal void ReplaceWith(IList<MenuBarItem> menuBarItems)
-		{
-			Clear();
```

Comment:
This was really supposed to just be temporary and might be causing some additional unknown weirdness.